docs/oid-info.md

```diff
@@ -58,11 +60,89 @@ the git ref that the workflow run was based upon.
 This specifies the username identity in the OtherName Subject Alternative Name, as
 defined by [RFC5280 4.2.1.6](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6).
 
+
+### 1.3.6.1.4.1.57264.1.8 | Source URI
+
+Should include a fully qualified source repository URL that the build was based
+on.
+
+### 1.3.6.1.4.1.57264.1.9 | Source Digest
+
+An immutable reference to a specific version of the source code that the build
+was based upon. Should include the digest type followed by the digest, e.g.
+`sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.10 | Source Ref
+
+The source ref that the build run was based upon. For example: `refs/head/main`.
+
+### 1.3.6.1.4.1.57264.1.11 | Source Identifier
+
+The stable identifier for the source repository the workflow was based upon.
+
+### 1.3.6.1.4.1.57264.1.12 | Source Owner URI
+
+Fully qualified URL for the owner of the source repository that the build was based
+on.
+
+### 1.3.6.1.4.1.57264.1.13 | Source Owner Identifier
+
+The stable identifier for the owner of the source repository that the workflow was based upon.
+
+### 1.3.6.1.4.1.57264.1.14 | Build Config URI
+
+A reference to the initiating build instructions.
+
+### 1.3.6.1.4.1.57264.1.15 | Build Config Digest
+
+An immutable reference to the specific version of the top-level build
+instructions. Should include the digest type followed by the digest, e.g.
+`sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.16 | Build Signer URI
+
+Reference to specific build instructions that are responsible for signing. Can be the same as Build Config URI. For example a reusable workflow in GitHub Actions or a Circle CI Orbs.
+
+Note this value would also be set in the Subject Alternative Name (SAN).
+
+### 1.3.6.1.4.1.57264.1.17 | Build Signer Digest
+
+An immutable reference to the specific version of the build instructions that is responsible for signing. Should include the digest type followed by the digest, e.g. `sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.18 | Build Trigger
+
+The event or action that triggered the build.
+
+### 1.3.6.1.4.1.57264.1.19 | Runner Environment
+
+For platforms to specify whether the build took place in platform-hosted cloud infrastructure or customer-hosted infrastructure. For example: `platform-hosted` and `self-hosted`.
+
 ## 1.3.6.1.4.1.57264.2 | Policy OID for Sigstore Timestamp Authority
 
 Not used by Fulcio. This specifies the policy OID for the [timestamp authority](https://github.com/sigstore/timestamp-authority)
 that Sigstore operates.
 
-<!-- References -->
+## Mapping OIDC token claims to Fulcio OIDs
```

Comment:
A proposed new claim: a version claim. I like the idea of sticking direct OIDC claims into the extension fields in the table. What I'm worried about is that the OID extensions are not versioned, and there is no way of allowing an OIDC provider to "update" the mapping (because we don't want to keep adding fields from this standardization). Having a Fulcio-integration "version" will help allow providers to update the table claims.

For e.g. if GH decides to add a sha prefix to the source digest, assuming that Fulcio is not sanitizing or modifying claims into this table, then that would be breaking to policy verifiers.

Reply:
Agreed, and IMO this is another good motivation for a single "composite" extension that wraps all of these claims as suggested in #981 (although that will need to happen with a breaking version, not here).

Reply:
Thought a bit more on this - Does a version that Fulcio maintains solve the problem of breaking policy verifiers? And furthermore, is that Fulcio's problem?

GH making a change to the format of the source digest would affect a verifier if they were directly consuming the OIDC token. This seems more like a problem for the OIDC providers, that they should version their tokens and we can then also include that version value in an extension. Fulcio isn't opinionated on each claim structure (at least that's what we decided after chatting - If it was opinionated, then we would need a version that fulcio maintains)

Adding a claim in a new extension should not be a breaking change, and clients should search for it and gracefully handle it not being present (verification policies can require the extension if it should be critical).

Reply:
Chatting with Asra, throwing around some ideas (sorry if this is not clear, just trying to get this down to remember this):

* Each extension becomes a list of structs of {version, value}. If we have to update a extension value, we add a new struct to the list, with an updated version, while maintaining the previous version-value. 
* We must always operate in append-only, else we break verification for clients
* If we were to do something like #981, while being append-only, we'd have to duplicate the entire composite every time we'd do an update for any extension value for any CI provider
* The first bullet point here minimizes the duplications. 
* Another idea is a version-per-issuer, but this needs more thought

Reply:
Thinking more on this: I think we should not complicate client logic with versioning logic, and push back on integrators who will be making version breaking changes to their  OIDC tokens. In a future world, I'd like OIDC tokens maybe to be versioned and for that to be represented in a claim, but I think Fulcio self-managing a versioning scheme would be pretty tough as an intermediate versioning layer between the OIDC provider and the client.

As in: let's push back on integrators who won't satisfy stable claims for the ones they are committing to tin the mapping.

Reply:
Leaving this open in case anyone has any other thoughts, but I think we're all in agreement that we won't attempt to do any versioning.